### ISSUE TITLE
Implement training stage of training job

### ISSUE THREAD

**Issue by leoll2** [open-edge-platform/training_extensions#4868]
The "training" step is responsible for executing the actual training loop to fine-tune the model (forward / backward passes and evaluation on validation set). At the end of the loop, this step should also convert the resulting model to OpenVINO IR.

To instantiate the `OTXDataModule`: https://github.com/open-edge-platform/training_extensions/blob/33f43c3dbb4c5d9b62293b8bc1999f6e91e550bb/library/src/otx/data/module.py#L225

To instantiate the `OTXEngine`: https://github.com/open-edge-platform/training_extensions/blob/33f43c3dbb4c5d9b62293b8bc1999f6e91e550bb/library/src/otx/backend/native/engine.py#L78

### Sample training script

```python
train_otx_dataset = OTXMulticlassClsDataset(
    dm_subset=train_dm_dataset,
    transforms=[...],  # Setup transforms based on augmentation params
)

data_module = OTXDataModule.from_otx_datasets(
    train_dataset=train_otx_dataset,
    val_dataset=val_otx_dataset,
    test_dataset=test_otx_dataset,
    train_subset=SubsetConfig(batch_size=2),
)

engine = OTXEngine(
    model="efficientnet_b0",
    data=data_module,
)

engine.train(max_epochs=1)
```
